hybrasyl/Game.cs

```diff
@@ -191,36 +217,47 @@ public static HybrasylConfig GatherConfig()
 
         public static void Main(string[] args)
         {
-            //log4net configure
-            var logRepository = LogManager.GetRepository(Assembly.GetEntryAssembly());
-            XmlConfigurator.Configure(logRepository, new FileInfo("Log4Net.config"));
-
             // Make our window nice and big
             //Console.SetWindowSize(140, 36);  //Removed for cross-platform compatibility
-            LogLevel = Hybrasyl.Constants.DEFAULT_LOG_LEVEL;
             Assemblyinfo = new AssemblyInfo(Assembly.GetEntryAssembly());
 
-            
+            // Set our exit handler
+            AppDomain.CurrentDomain.ProcessExit += new EventHandler(CurrentDomain_ProcessExit);
+
             Constants.DataDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), "Hybrasyl");
 
             if (!Directory.Exists(Constants.DataDirectory))
             {
-                Logger.InfoFormat("Creating data directory {0}", Constants.DataDirectory);
-                try
-                {
-                    // Create the various directories we need
-                    Directory.CreateDirectory(Constants.DataDirectory);
-                    Directory.CreateDirectory(Path.Combine(Constants.DataDirectory, "maps"));
-                    Directory.CreateDirectory(Path.Combine(Constants.DataDirectory, "scripts"));
-                    Directory.CreateDirectory(Path.Combine(Constants.DataDirectory, "world"));
-                }
-                catch (Exception e)
-                {
-                    Logger.ErrorFormat("Can't create data directory: {0}", e.ToString());
-                    return;
-                }
+                Log.Information("Creating data directory {Directory}", Constants.DataDirectory);
+            }
+            try
+            {
+                // Ensure at least the world and logs directory exist 
+                Directory.CreateDirectory(Constants.DataDirectory);
+                Directory.CreateDirectory(Path.Combine(Constants.DataDirectory, "world"));
+                Directory.CreateDirectory(Path.Combine(Constants.DataDirectory, "logs"));
             }
-            
+            catch (Exception e)
+            {
+                Log.Fatal("Can't create data directory: {Directory}", e.ToString());
+                return;
+            }
+
+            // Configure logging 
+
+            // Default is info
+            LevelSwitch = new LoggingLevelSwitch();
+            LevelSwitch.MinimumLevel = LogEventLevel.Information;
+
+            // We log every LogType defined in our enumeration to its own file. Only the "general" type is sent to the console.
+            var log = new LoggerConfiguration().MinimumLevel.ControlledBy(LevelSwitch).Enrich.WithThreadId().Enrich.WithExceptionData().
+                WriteTo.Map("LogType", "General", (name, wt) => wt.File($"{Path.Combine(Constants.DataDirectory, "logs")}/log-{name}.log", rollingInterval: RollingInterval.Day)).

```

Comment:
does rolling interval maintain previous logs with a datestamp attached? if no, that'd probably be beneficial.

Reply:
may also want to set a maximum logfile size/length.

Reply:
Yes, it maintains them, no removal is done.
The logs end up like: `log-General20190823.log` though now that I think about it they could just be called `General-20190823.log` so i'll update that

Reply:
Name updated, rotated automatically if size exceeds 1gb, 90 days of logs kept